common/djangoapps/third_party_auth/middleware.py

```diff
@@ -31,6 +31,12 @@ def get_redirect_uri(self, request, exception):
         if auth_entry and auth_entry in pipeline.AUTH_DISPATCH_URLS:
             redirect_uri = pipeline.AUTH_DISPATCH_URLS[auth_entry]
 
+            # PhilU check for redirection in popup based registration
+            register_origin_url = request.session.get('next')
```

Comment:
@saad-waqar can we set next to query string params?

Reply:
Qasim bhai it would mean having the same url in next as the original url.